### ISSUE TITLE
Recent update changed autocompletion for aliases

### ISSUE THREAD

**Issue by parsonsmatt** [robbyrussell/oh-my-zsh#4064]
The recent update to oh-my-zsh seems to have changed autocompletion behavior for aliases.

The `git` plugin gives the alias [`gco`](https://github.com/robbyrussell/oh-my-zsh/blob/master/plugins/git/git.plugin.zsh#L77) for `git checkout`. When using `git checkout`, autocompletion works as you'd expect (showing available branches, choosing available branch that matches). When using `gco`, it attempts to autocomplete files/folders in the current directory.

Same for `gc` -> `git commit`: `gc` autocompletion only uses the files in the current directory, whereas `git commit` will autocomplete among files that have been modified.

In prior versions, the aliases had the same completion as the full commands.

**Comment by ncanceill** [comment]
Hi, and thanks for reporting.

This should not happen, but I have no time to try and reproduce in the next few days. We recently removed the custom completion declarations from the plugin since they should have been useless — theoretically, Zsh is smart enough to look up aliases and load the proper completion...

I will get back to you when I can reproduce the bug.

@blueyed any opinion?

**Comment by blueyed** [comment]
There is a Zsh option for this.  Check your output of `setopt`.

**Comment by parsonsmatt** [comment]
setopt output:

alwaystoend
autocd
autopushd
nobgnice
cdablevars
completealiases
completeinword
correct
extendedhistory
noflowcontrol
histexpiredupsfirst
histignorealldups
histignoredups
histignorespace
histreduceblanks
histverify
nohup
ignoreeof
incappendhistory
interactive
interactivecomments
nolistbeep
localoptions
localtraps
login
longlistjobs
monitor
promptsubst
pushdignoredups
pushdminus
sharehistory
shinstdin
zle

Anything in here that isn't right?

I did try checking out an earlier version of the repo, reloading oh-my-zsh, and then autocompletion worked. When I tried to bisect, though, it didn't *stop* working. I'm going to attempt to identify the commit that introduced the bug but it might take a bit.

**Comment by blueyed** [comment]
completealiases

Where do you set it?

**Comment by parsonsmatt** [comment]
According to `ag`, that's not mentioned anywhere in my home directory at
all, including the oh-my-zsh dir and my dotfiles repo. Doing `unsetopt
completealiases` does fix the issue though.

If no one else is reporting trouble with this, I'm fine closing the issue
and calling it a quirk on my own system.

Matt Parsons

**Comment by blueyed** [comment]
> that's not mentioned anywhere in my home directory

Have you maybe just upgraded omz? (it was reverted in 8883ace).
And did you search with `ag -i 'complete_?alias'`?

**Comment by parsonsmatt** [comment]
So, I did a git bisect again, and it found the commit that introduced the problem. I lost the info though and would have to refind it.

On your suggestion, I also looked a bit more closely at the `ag` command line flags, and found that it wasn't searching my dotfiles. So then I did [this](https://github.com/parsonsmatt/dotfiles/commit/25f85e167d26cb3f223fec4da31a55c8d7700ba1) and everything is fine :o that's what I get for using someone else's dotfiles I guess.

Thanks for helping me out!
